adapters/kubernetes_generic/kubernetes_generic_connect.php

```diff
@@ -162,61 +160,76 @@ public function send($origin, $cmd)
         }
 
         // TODO TEST validation ACTION[] fields
-        $curl_cmd = "curl --tlsv1.2 -i -sw '\nHTTP_CODE=%{http_code}' --connect-timeout {$delay} --max-time {$delay} -X {$action[0]} {$token} -H \"Content-Type: application/json\" -k '{$action[2]}'";
+
+        $delay = EXPECT_DELAY / 1000;
+        $http_header = array();
+        $http_header[] = 'Content-Type: application/json';
+        // $curl_cmd is used for logs
         if ($kube_auth_method == "KUBERNETES" || $kube_auth_method == "EKS") {
-            $curl_cmd = "curl --tlsv1.2 -i -sw '\nHTTP_CODE=%{http_code}' --connect-timeout {$delay} --max-time {$delay} -X {$action[0]} --header \"Authorization: Bearer {$token}\" -H \"Content-Type: application/json\" -k '{$action[2]}'";
+            $curl_cmd = "curl --tlsv1.2 -i -sw '\nHTTP_CODE=%{http_code}' --connect-timeout {$delay} --max-time {$delay} -X {$action[0]} --header \"Authorization: Bearer {$token}\" -H \"Content-Type: application/json\" -k '{$url}'";
+            $http_header[] = "Authorization: Bearer {$token}";
+        } else {
+            $curl_cmd = "curl --tlsv1.2 -i -sw '\nHTTP_CODE=%{http_code}' --connect-timeout {$delay} --max-time {$delay} -X {$action[0]} --header \"X-Auth-Token: {$token}\" -H \"Content-Type: application/json\" -k '{$url}'";
+            $http_header[] = "X-Auth-Token: {$token}";
         }
 
         if (isset($action[3]) && ($kube_auth_method == "KUBERNETES" || $kube_auth_method == "EKS")) {
             $curl_cmd .= " -d '{$action[3]}'";
+            $post_fields = $action[3];
         }
 
-        echo "{$cmd} for endPoint {$action[1]}\n";
+        echo "{$cmd} for endPoint {$url}\n";
+
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
+        curl_setopt($ch, CURLOPT_HEADER, true);
+        switch($action[0]) {
+          case 'POST':
+            curl_setopt($ch, CURLOPT_POST, true);
+            curl_setopt($ch, CURLOPT_POSTFIELDS, $post_fields);
+            break;
+
+          case 'PUT':
+            curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "PUT");
+            curl_setopt($ch, CURLOPT_POSTFIELDS, $post_fields);
+            break;
+
+          case 'DELETE':
+            curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "DELETE");
+            break;
+
+          default:
+            // GET
+            break;
+        }
+        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+        curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);
+        curl_setopt($ch, CURLOPT_SSLVERSION, CURL_SSLVERSION_MAX_TLSv1_2);
+        curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, $delay);
+        curl_setopt($ch, CURLOPT_TIMEOUT, $delay);
+        curl_setopt($ch, CURLOPT_HTTPHEADER, $http_header);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        $ret = curl_exec($ch);
+        if ($ret === false) {
```

Comment:
Instead of testing $ret I usually do:
`
      if (curl_errno($ch))
      {
          sms_log_error("curl error: " . curl_error($ch));
      }
`